app/services/group_stage_service.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class GroupStageService
+  def self.generate_group_stage(groups)
+    return false if groups.length.zero?
+
+    average_group_size = (groups.flatten.length.to_f / groups.length.to_f)
+    if (average_group_size % 1).zero?
+      groups = groups.map { |group| get_group_object_from(group) }
+      group_stage = Stage.new level: -1, groups: groups
+      group_stage
+    else
+      false
+    end
+  end
+
+  def self.get_group_object_from(team_array)
+    matches = generate_all_matches_between team_array
+    group = Group.new matches: matches
+    group
```

Comment:
this line can be removed by not assigning the statement above to a var (just writing it as is)